android/src/test/java/vn/luongvo/kmm/survey/android/test/CoroutineTestRule.kt

```diff
@@ -0,0 +1,38 @@
+package vn.luongvo.kmm.survey.android.test
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.*
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import vn.luongvo.kmm.survey.android.util.DispatchersProvider
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class CoroutineTestRule : TestWatcher() {
+
+    internal val testDispatcher: TestCoroutineDispatcher = TestCoroutineDispatcher()
+
+    override fun starting(description: Description?) {
+        super.starting(description)
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    override fun finished(description: Description?) {
+        super.finished(description)
+        Dispatchers.resetMain()
+        testDispatcher.cleanupTestCoroutines()
+    }
+
+    val testDispatcherProvider = object : DispatchersProvider {
+        override val io: CoroutineDispatcher
+            get() = testDispatcher
+        override val main: CoroutineDispatcher
+            get() = testDispatcher
+        override val default: CoroutineDispatcher
+            get() = testDispatcher
+    }
+}
+
+@OptIn(ExperimentalCoroutinesApi::class)
+fun CoroutineTestRule.runBlockingTest(block: suspend TestCoroutineScope.() -> Unit) {
```

Comment:
I think we need to improve this one as `runBlockingTest` is deprecated.

Reply:
@suho Sure, I will migrate to get rid of `runBlockingTest` when working on an actual test of a VM. Let me note this issue down and ignore fixing in this PR 🙏